vtex-sites/nextjs.store
# Feat: Add css modules, tokens and refactor `Search` components

Signed-off-by: Arthur Andrade <arthurfelandrade@gmail.com>

## What's the purpose of this pull request?

## How does it work?

This PR renames the components from Search feature to follow the Pattern `Search[ComponentName]`, and relocates the components to group in search folder, as that's components share the same states provided for `useInputSearch`  and belong to the same feature.

The refactored components are: 

<table>
<tr>
	<td>Before</td>
	<td>After</td>
<tr>
	<td>src/components/common/SearchInput
	<td>src/components/search/SearchInput
<tr>
	<td>src/components/search/History
	<td>src/components/search/SearchHistory
<tr>
	<td>src/components/search/Suggestions
	<td>src/components/search/SearchDropdown
<tr>
	<td>src/components/search/SuggestionProductCard
	<td>src/components/search/SearchProductCard
<tr>
	<td>src/components/ui/Search/Suggestions
	<td>src/components/search/SearchSuggestions
<tr>
	<td>src/components/search/TopSuggestions/index.ts
	<td>src/components/search/SearchTop/index.ts
</table>

Also, this PR adds tokens to `SearchHistory`, but when it was done, it was perceived that the `SearchHistory` style is shared with `SearchTop`, `SearchDropdown` and `SearchSuggestions`, so file src/components/search/search-common.module.scss was created to be added to these 4 components.

To finish, storybook documentation was created for all search components, with an [Overview documentation for the feature](https://nextjs-store-storybook-git-fs-381-add-new-toke-4a92b4-faststore.vercel.app/?path=/docs/features-search-overview--overview) using API mocks add-on.

## How to test it?

To test, you can check the feature [Search in storybook documentation](https://nextjs-store-storybook-git-fs-381-add-new-toke-4a92b4-faststore.vercel.app/?path=/docs/features-search-overview--overview) and check the style and behaviours of the feature in [preview](https://sfj-30e6996--nextjs.preview.vtex.app/).

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**

**PR Description**

**Documentation**
